src/query/pipeline/sources/src/processors/sources/input_formats/input_format_text.rs

```diff
@@ -135,6 +110,58 @@ impl<T: InputFormatTextBase> InputFormat for InputFormatText<T> {
     fn exec_stream(&self, ctx: Arc<InputContext>, pipeline: &mut Pipeline) -> Result<()> {
         InputFormatTextPipe::<T>::execute_stream(ctx, pipeline)
     }
+
+    async fn get_splits(
+        &self,
+        plan: &CopyIntoPlan,
```

Comment:
It's better not to rely on logical plan directly. How about passing `stage_info` and `files`?

BTW, what will happen if `files` is empty?

Reply:
this CopyIntoPlan is not logical plan. I may rename it later.

---

> BTW, what will happen if files is empty?

```
2022-09-26T14:04:00.726888Z  INFO databend_query::interpreters::interpreter_copy_v2: matched copy unduplicate files: []
2022-09-26T14:04:00.727168Z  INFO databend_query::interpreters::interpreter_query_log: {"log_type":2,"handler_type":"... }

```

Reply:
Ok, I got it. There is a plan to rewrite copy in a streaming way: https://github.com/datafuselabs/databend/issues/7889

Can you retry this function to accept a `impl Stream<Result<StageFile>>` instead of files? (The `files` could contain 200W entries)

It's OK to keep this PR AS-IS and do this change in the future.

Reply:
got it.